### PR TITLE
[Foxy backport] Add PoseWithCovarianceStamped transform support

### PR DESCRIPTION
**Pull request by nnmm**
Backport of https://github.com/ros2/geometry2/pull/312

I compiled and tested ROS2 Foxy with this change locally. A few tests failed, but none that seemed related to this change. API and ABI stability should be guaranteed because functions were only added, not modified.

### CODE REVIEW

**Comment by nnmm** [comment]
@clalancette Do I need to request a reviewer? Sorry, this is my first contribution :)

**Comment by nnmm** [comment]
cc @dejanpan

**Review by clalancette — APPROVED** [review]
Seems reasonable to me, I'll run CI on it next.

**Comment by clalancette** [comment]
CI:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/12897/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/7851/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/10613/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/12839/)

**Comment by nnmm** [comment]
Thanks, awesome! @clalancette do you know roughly when the Foxy patch release with this change will be released?

**Comment by clalancette** [comment]
> Thanks, awesome! @clalancette do you know roughly when the Foxy patch release with this change will be released?

We'll still have to do a release into Foxy, and then it will make the next sync.  That's typically every 2 weeks, though it sometimes gets delayed for one reason or another.

**Comment by nnmm** [comment]
Okay, does "release into Foxy" mean a PR like this one? https://github.com/ros/rosdistro/pull/26627

I'm a bit confused by this [project board](https://github.com/orgs/ros2/projects/28) which has individual PRs marked as "need release". Also not sure what sync means, but I'm assuming no action from me is needed anymore.

**Comment by clalancette** [comment]
> Okay, does "release into Foxy" mean a PR like this one? [ros/rosdistro#26627](https://github.com/ros/rosdistro/pull/26627)

Yes, that's correct.  I've opened one for Foxy now (and its been merged): https://github.com/ros/rosdistro/pull/27177

That means that the buildfarm is currently building the package, and any packages that depend on it.  Depending on how far down in the stack that is, that could take anywhere from a few minutes to several hours.

> I'm a bit confused by this [project board](https://github.com/orgs/ros2/projects/28) which has individual PRs marked as "need release".

That's right, things in that project board that are in the "Needs Release" column need a release like I described above.

> Also not sure what sync means

In short: we are continually building packages for the ROS distributions with the latest fixes.  These all land in a "testing" repository.  Every few weeks, we do a sync from the testing repository out to the main repositories, and hence make it available to everyone.

> but I'm assuming no action from me is needed anymore.

That's correct.  Assuming that it all builds fine on the buildfarm now, it will be in the next sync.
